src/udev/udev-builtin-path_id.c

```diff
@@ -200,6 +202,16 @@ static sd_device *handle_scsi_sas(sd_device *parent, char **path) {
                 /* Get expander's address */
                 if (sd_device_get_sysattr_value(expander_sasdev, "sas_address", &sas_address) < 0)
                         return NULL;
+
+                /* Use expander-side phy id */
+                if (sd_device_get_sysname(port, &sysname) < 0)
+                        return NULL;
+                if (!snprintf_ok(phyname, sizeof phyname, "phy-%s", sysname + strlen("port-")))
```

Comment:
Please use `STRLEN()` macro here.

Is it not necessary to check that `sysname` starts with "port-"?

Reply:
> Please use `STRLEN()` macro here.
Sure.

> Is it not necessary to check that `sysname` starts with "port-"?
Better not to assume it, I'll add a check.

Reply:
No, there's no reason to use STRLEN. STRLEN was only added for VLAs (or more exactly, to avoid having a VLA: STRLEN allows a normal fixed-length array to be used). In code like this, gcc and clang optimize `strlen` to a fixed constant, so there's no difference.

Reply:
Oh, I did not know that. In many place of udev code, `STRLEN()` macro is used for cutting the head of string. I will drop such use cases in a PR. Thanks.

Reply:
`STRLEN` is not *wrong*, it still works fine. It's just unnecessary in those cases. (So yeah, dropping them is fine, but it's nothing urgent.)